crypto/dilithium/p_pqdsa_test.cc

```diff
@@ -1108,232 +1108,323 @@ TEST_P(PQDSAParameterTest, KAT) {
-  ASSERT_TRUE(EVP_PKEY_keygen(ctx1.get(), &raw1));
-  bssl::UniquePtr<EVP_PKEY> pkey1(raw1);
+  bssl::UniquePtr<EVP_PKEY> pkey1(generate_key_pair(nid));
 
   // Generate second keypair
-  bssl::UniquePtr<EVP_PKEY_CTX> ctx2(EVP_PKEY_CTX_new_id(EVP_PKEY_PQDSA, nullptr));
-  ASSERT_TRUE(ctx2);
-  ASSERT_TRUE(EVP_PKEY_CTX_pqdsa_set_params(ctx2.get(),nid));
-  ASSERT_TRUE(EVP_PKEY_keygen_init(ctx2.get()));
-  EVP_PKEY *raw2 = nullptr;
-  ASSERT_TRUE(EVP_PKEY_keygen(ctx2.get(), &raw2));
-  bssl::UniquePtr<EVP_PKEY> pkey2(raw2);
+  bssl::UniquePtr<EVP_PKEY> pkey2(generate_key_pair(nid));
 
   // Compare keys
   EXPECT_EQ(0, EVP_PKEY_cmp(pkey1.get(), pkey2.get()));
 }
 
 TEST_P(PQDSAParameterTest, KeySize) {
-  // Test the key size of MLDSA key is as expected
+  // Test the key size of PQDSA key is as expected
   int nid = GetParam().nid;
   int pk_len = GetParam().public_key_len;
   int sig_len = GetParam().signature_len;
 
-  // generate an MLDSA keypair
-  bssl::UniquePtr<EVP_PKEY_CTX> ctx(EVP_PKEY_CTX_new_id(EVP_PKEY_PQDSA, nullptr));
-  ASSERT_TRUE(ctx);
-  ASSERT_TRUE(EVP_PKEY_CTX_pqdsa_set_params(ctx.get(),nid));
-  ASSERT_TRUE(EVP_PKEY_keygen_init(ctx.get()));
-  EVP_PKEY *raw = nullptr;
-  ASSERT_TRUE(EVP_PKEY_keygen(ctx.get(), &raw));
-  bssl::UniquePtr<EVP_PKEY> pkey(raw);
-
+  bssl::UniquePtr<EVP_PKEY> pkey(generate_key_pair(nid));
   EXPECT_EQ(sig_len, EVP_PKEY_size(pkey.get()));
   EXPECT_EQ(8*(pk_len), EVP_PKEY_bits(pkey.get()));
 }
 
-TEST_P(PQDSAParameterTest, NewKeyFromBytes) {
-  // Test the generation of a MLDSA key from bytes
+TEST_P(PQDSAParameterTest, RawFunctions) {
+  // ---- 1. Setup phase: generate PQDSA EVP KEY ----
   int nid = GetParam().nid;
+  bssl::UniquePtr<EVP_PKEY> pkey(generate_key_pair(GetParam().nid));
+
+  // ---- 2. Test the extraction of public and private key from a PKEY ----
   size_t pk_len = GetParam().public_key_len;
   size_t sk_len = GetParam().private_key_len;
+  std::vector<uint8_t> pk(pk_len);
+  std::vector<uint8_t> sk(sk_len);
+
+  ASSERT_TRUE(EVP_PKEY_get_raw_public_key(pkey.get(), pk.data(), &pk_len));
+  ASSERT_TRUE(EVP_PKEY_get_raw_private_key(pkey.get(), sk.data(), &sk_len));
+  EXPECT_EQ(pk_len, GetParam().public_key_len);
+  EXPECT_EQ(sk_len, GetParam().private_key_len);
+  CMP_VEC_AND_PKEY_PUBLIC(pk, pkey, pk_len);
+  CMP_VEC_AND_PKEY_SECRET(sk, pkey, sk_len);
+
+  // ---- 3. Test getting public/private key sizes ----
+  pk_len = 0;
+  sk_len = 0;
+  ASSERT_TRUE(EVP_PKEY_get_raw_public_key(pkey.get(), nullptr, &pk_len));
+  ASSERT_TRUE(EVP_PKEY_get_raw_private_key(pkey.get(), nullptr, &sk_len));
+  EXPECT_EQ(pk_len, GetParam().public_key_len);
+  EXPECT_EQ(sk_len, GetParam().private_key_len);
+
+  // ---- 4. Test creating PKEYs from raw data ----
+  bssl::UniquePtr<EVP_PKEY> public_pkey(EVP_PKEY_pqdsa_new_raw_public_key(nid, pkey->pkey.pqdsa_key->public_key, pk_len));
+  bssl::UniquePtr<EVP_PKEY> private_pkey(EVP_PKEY_pqdsa_new_raw_private_key(nid, pkey->pkey.pqdsa_key->private_key, sk_len));
```

Comment:
Lines length

Reply:
fixed in [f7d1dad](https://github.com/aws/aws-lc/pull/2062/commits/f7d1dadb64f54b7695d871a88d4dfacf42048637)